coriolis/osmorphing/base.py

```diff
@@ -572,3 +572,266 @@ def _set_grub2_console_settings(self, consoles=None, speed=None,
+                        break
+
+            if not mac_address:
+                self._event_manager.warn(
+                    "mac-address not defined and no matching IP found,"
+                    "skipping: %s" % ifcfg_file)
+                continue
+
+            name = ifcfg.get("DEVICE")
+            if not name:
+                # Get the name from the config file
+                name = re.match("^.*/ifcfg-(.*)", ifcfg_file).groups()[0]
+            net_ifaces_info[name] = mac_address
+        return net_ifaces_info
+
+    def _get_nmconnection_net_ifaces_info(self, nmconnection_files, nics_info):
+        net_ifaces_info = dict()
+
+        for nmconn_file, nmconn in nmconnection_files:
+            mac_address = nmconn.get("mac-address")
+            nm_ip_addresses = []
+            for key, value in nmconn.items():
+                if key.lower().startswith("address") and value:
+                    for addr in value.split(','):
+                        addr = addr.strip()
+                        ip = addr.split('/')[0] if '/' in addr else addr
+                        nm_ip_addresses.append(ip)
+
+            if not mac_address and nm_ip_addresses:
+                for ip in nm_ip_addresses:
+                    for nic in nics_info:
+                        nic_ip_addresses = nic.get('ip_addresses')
+                        if ip in nic_ip_addresses:
+                            mac_address = nic.get('mac_address')
+                            break
+
+            if not mac_address:
+                self._event_manager.warn(
+                    "mac-address not defined and no matching IP found,"
+                    "skipping: %s" % nmconn_file)
+                continue
+
+            name = nmconn.get("connection", {}).get("id")
+            if not name:
+                name = re.match(
+                    r"^.*/(.*)\.nmconnection$", nmconn_file).groups()[0]
+
+            net_ifaces_info[name] = mac_address
+
+        return net_ifaces_info
+
+    def _set_static_config(self, nics_info) -> None:
+        net_ifaces_info = dict()
+        network_scripts_path = "etc/sysconfig/network-scripts"
+        netplan_base = "etc/netplan"
+        ifaces_file = "etc/network/interfaces"
+        nmconnection_file = "etc/NetworkManager/system-connections"
+
+        if self._test_path(nmconnection_file):
+            nm_files = self._get_nmconnection_files(nmconnection_file)
+            if nm_files:
+                nmconnection_ethernet = self._get_keyfiles_by_type(
+                    "ethernet", nmconnection_file)
+                if nmconnection_ethernet:
+                    net_ifaces_info = self._get_nmconnection_net_ifaces_info(
+                        nmconnection_ethernet, nics_info)
+                    LOG.info("Using NetworkManager connection files for "
+                             "configuring static IP")
+
+        if not net_ifaces_info and self._test_path(ifaces_file):
```

Comment:
Why not just use `elif` instead of adding `not net_ifaces_info`? 

This will not work anymore if you switch to the new design. You should specifically check if the path is populated with valid config files.

Reply:
This feels like an excellent opportunity to apply the Factory Design Pattern. We'll discuss the design details.